src/python/pants/source/source_root.py

```diff
@@ -29,15 +31,20 @@ class AllSourceRoots(Collection[SourceRoot]):
 class SourceRoots:
     """An interface for querying source roots."""
 
-    def __init__(self, source_root_config: "SourceRootConfig") -> None:
+    def __init__(
+        self,
+        roots: List[str],
```

Comment:
Better for this to be `Iterable[str]`. You end up converting it to a `set`, so there's no reason to confine this.